timemachine/ff/handlers/nonbonded.py

```diff
@@ -15,6 +15,9 @@
 from timemachine.ff.handlers.utils import match_smirks as rd_match_smirks
 from timemachine.ff.handlers.utils import sort_tuple
 
+AM1_CHARGE_CACHE = "AM1Cache"
+AM1_SMIRK_MATCH_CACHE = "AM1BondMatchCache"
```

Comment:
Nit: naming -- these smirks matches are not strictly AM1-derived. Perhaps `BCC_SMIRK_MATCH_CACHE` or similar.

(In general, the list of smirks matches will depend on which force field is loaded -- not sure if this design needs to be defensive against populating the cache with one force field, then using the cache with a different force field. Tangent: An analogous consideration also applies to how the charge cache is implemented: https://github.com/proteneer/timemachine/pull/571#discussion_r788307734 )

Reply:
In this context, it looks like we are using vanilla `AM1` not `AM1BCC` or `AM1BCCELF10`. Should we be using the BCC charges?

Thats a good question, will need to think about that.

Reply:
I may have misunderstood -- I thought the bond smirks matches were defined by a file like https://github.com/proteneer/timemachine/blob/5d8f3099aafa1cf7def0585525aaf4de73c23343/timemachine/ff/params/smirnoff_1_1_0_ccc.py#L343 , not by AM1

Reply:
You are right they are, sorry had thought we were using the AM1 charges in the function but we are not. Perhaps just `BOND_SMIRK_MATCH_CACHE`, as really the function doesn't care about the origins of the smirks?

Reply:
(For example, if we parameterize mol using the list of patterns in https://github.com/proteneer/timemachine/blob/5d8f3099aafa1cf7def0585525aaf4de73c23343/tests/test_handlers.py#L298 , we will populate this cache in an arbitrary, not AM1-derived way. If we then try to parameterize the same mol object with another force field with a different list of patterns, we will attempt to load from the cache, but the cache will be stale.)

Reply:
>Perhaps just BOND_SMIRK_MATCH_CACHE, as really the function doesn't care about the origins of the smirks?

That seems a better name.

(There's a separate issue of when the new cache might become stale, but this doesn't have to be addressed here -- similar to the current level of safety.)

Reply:
Renamed in [21039c3](https://github.com/proteneer/timemachine/pull/640/commits/21039c3556e0ba96419309922e6b5c41f8411330)